clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -194,10 +194,13 @@ public Password getPassword(String key) {
         Map<String, Object> result = new RecordingMap<>(values(), prefix, true);
         for (Map.Entry<String, ?> entry : originals.entrySet()) {
             if (entry.getKey().startsWith(prefix) && entry.getKey().length() > prefix.length()) {
-                String keyWithNoPrefix = entry.getKey().substring(prefix.length());
+                String prefixedName = entry.getKey();
+                String keyWithNoPrefix = prefixedName.substring(prefix.length());
                 ConfigDef.ConfigKey configKey = definition.configKeys().get(keyWithNoPrefix);
-                if (configKey != null)
-                    result.put(keyWithNoPrefix, definition.parseValue(configKey, entry.getValue(), true));
+                if (configKey != null) {
+                    result.put(configKey.name, ConfigDef.
```

Comment:
Why are we using the config key name as the key instead of `keyWithNoPrefix`?

Reply:
`configKey.name` is always the true name, whereas `keyWithNoPrefix` may be a deprecated synonym.  We want the `values` map to contain true names as keys, not deprecated synonyms.

This is consistent with how the constructor of `AbstractConfig` invokes `ConfigDef#parse` to set  up `AbstractConfig#values`.  It should always map true names to values.  Its key set should not contain deprecated names.

Reply:
The intention is that deprecated key strings are not referred to by anyone in the code ever-- except when defining `ConfigKey` objects.  All other parts of the code use the most current key names.